storey/sources.py

```diff
@@ -80,27 +81,32 @@ class FlowController(FlowControllerBase):
     To be used from a synchronous context.
     """
 
-    def __init__(self, emit_fn, await_termination_fn, key_field: Optional[str] = None,
+    def __init__(self, emit_fn, await_termination_fn, has_complete_in_flow, key_field: Optional[str] = None,
                  time_field: Optional[str] = None):
         super().__init__(key_field, time_field)
         self._emit_fn = emit_fn
         self._await_termination_fn = await_termination_fn
+        self.return_awaitable_result = has_complete_in_flow
 
     def emit(self, element: object, key: Optional[str] = None, event_time: Optional[datetime] = None,
-             return_awaitable_result: bool = False):
+             return_awaitable_result: Optional[bool] = None):
         """Emits an event into the associated flow.
 
         :param element: The event data, or payload. To set metadata as well, pass an Event object.
         :param key: The event key (optional)
         :param event_time: The event time (default to current time, UTC).
-        :param return_awaitable_result: Whether an AwaitableResult object should be returned. Defaults to False.
-        :type return_awaitable_result: boolean
+        :param return_awaitable_result: Deprecated! will return awaitable result if Complete() is part of flow
+        :type return_awaitable_result: boolean.
 
-        :returns: AsyncAwaitableResult if return_awaitable_result is True. None otherwise.
+        :returns: AsyncAwaitableResult if Complete() is part of the flow. None otherwise.
         """
+        if return_awaitable_result is not None:
+            warnings.warn("return_awaitable_result is deprecated. awaitable result will be returned if there is Complete() in the flow",
```

Comment:
An awaitable result object will be returned if a Complete step appears in the flow.

Reply:
Similarly in other strings.